provider/vespa.go

```diff
@@ -0,0 +1,132 @@
+package provider
+
+import (
+    "context"
+    "encoding/json"
+    "fmt"
+    "io/ioutil"
+    "net/http"
+    "net/url"
+
+  
+    pc "github.com/featureform/provider/provider_config"
+   pt "github.com/featureform/provider/provider_type"
+)
+
+type VespaVectorStore struct {
+    url string
+}
+
+type VectorStore interface {
+ CreateIndex(feature, variant string, vectorType VectorType) (VectorStoreTable, error)
+ DeleteIndex(feature, variant string) error
+ OnlineStore
+}
+
+type VectorStoreTable interface {
+ OnlineStoreTable
+ Nearest(feature, variant string, vector []float32, k int32) ([]string, error)
+}
```

Comment:
@neelshah2409, you'll need to remove these interface definitions as they are already declared in `online.go`, which means the project won't compile.

Perhaps I should've been clearer in my first comment, but what you'll need to do is ensure your `VespaVectorStore` implements the `VectorStore` interface.